sde_collections/sinequa_api.py

```diff
@@ -94,11 +89,40 @@ def query(self, page: int, collection_config_folder: str = "") -> Any:
         }
 
         if collection_config_folder:
-            if self.server_name in ["lis_server", "lrm_dev_server", "lrm_qa_server"]:
+            if self.server_name in ["xli", "lrm_dev", "lrm_qa"]:
                 payload["query"]["advanced"]["collection"] = f"/scrapers/{collection_config_folder}/"
             else:
                 payload["query"]["advanced"]["collection"] = f"/SDE/{collection_config_folder}/"
 
         response = self.process_response(url, payload)
 
         return response
+
+    def sql_query(self, sql: str) -> Any:
+        """Executes an SQL query on the configured server using token-based authentication."""
+        if not self.token:
+            raise ValueError("You must have a token to use the SQL endpoint")
+
+        url = f"{self.base_url}/api/v1/engine.sql"
+        headers = {"Content-Type": "application/json", "Authorization": f"Bearer {self.token}"}
+        payload = json.dumps(
+            {
+                "method": "engine.sql",
+                "sql": sql,
+                "pretty": True,
+                "log": False,
+                "output": "json",
+                "resolveIndexList": "false",
+                "engines": "default",
+            }
+        )
+        try:
+            response = requests.post(url, headers=headers, data=payload, timeout=10)
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"API request failed: {str(e)}")
+
+    def get_full_texts(self, collection_config_folder: str) -> Any:
```

Comment:
can you add an example output here, with all the fields and a fake value:
For example
{'full_text':string, 'collection_name':string}

Reply:
An example has been added. This is subsequently processed/extracted by the fetch_and_update_full_text function defined in tasks.py